src/controllers/csv-processor.ts

```diff
@@ -101,16 +102,75 @@ function validateParams(page_number: number, max_page_number: number, page_size:
     return errors;
 }
 
-export const uploadCSV = async (buff: Buffer, dataset: Dataset): Promise<UploadDTO | UploadErrDTO> => {
+export const moveCSVFromBlobStorageToDatalake = async (dataset: Dataset): Promise<UploadDTO | UploadErrDTO> => {
+    const blobStorageService = new BlobStorageService();
     const dataLakeService = new DataLakeService();
+    const datafiles = await dataset.datafiles;
+    const datafile: Datafile | undefined = datafiles
+        .filter((filterfile: Datafile) => filterfile.draft === false)
+        .sort(
+            (first: Datafile, second: Datafile) =>
+                new Date(second.creationDate).getTime() - new Date(first.creationDate).getTime()
+        )
+        .shift();
+    const dto = await datasetToDatasetDTO(dataset);
+    if (datafile) {
+        try {
+            logger.debug(`Moving file ${datafile.id} from blob storage to datalake`);
+            const buff: Buffer = await blobStorageService.readFileToBuffer(`${datafile.id}.csv`);
+            await dataLakeService.uploadFile(`${datafile.id}.csv`, buff);
+            datafile.draft = false;
+            await datafile.save();
+            await blobStorageService.deleteFile(`${datafile.id}.csv`);
+            return {
+                success: true,
+                dataset: dto
+            };
+        } catch (err) {
+            logger.error(err);
+            return {
+                success: false,
+                dataset: dto,
+                errors: [
+                    {
+                        field: 'csv',
+                        message: [
+                            { lang: ENGLISH, message: t('errors.move_to_datalake', { lng: ENGLISH }) },
+                            { lang: WELSH, message: t('errors.move_to_datalake', { lng: WELSH }) }
+                        ],
+                        tag: { name: 'errors.move_to_datalake', params: {} }
+                    }
+                ]
+            };
+        }
+    } else {
+        return {
+            success: false,
+            dataset: dto,
+            errors: [
+                {
+                    field: 'csv',
+                    message: [
+                        { lang: ENGLISH, message: t('errors.no_csv', { lng: ENGLISH }) },
+                        { lang: WELSH, message: t('errors.no_csv', { lng: WELSH }) }
+                    ],
+                    tag: { name: 'errors.no_csv', params: {} }
+                }
+            ]
+        };
+    }
+};
+
+export const uploadCSVToBlobStorage = async (buff: Buffer, dataset: Dataset): Promise<UploadDTO | UploadErrDTO> => {
+    const blobStorageService = new BlobStorageService();
     const hash = createHash('sha256').update(buff).digest('hex');
     const datafile = Datafile.createDatafile(dataset, hash.toString(), 'BetaUser');
     const savedDataFile = await datafile.save();
     const dto = await datasetToDatasetDTO(dataset);
     if (buff) {
         try {
```

Comment:
Thats because this only deals with putting the file into Blob Storage.  We don't make any requests to the datalake at this point.  We are just storing the file the user supplied so that we can render it back to the user so they can preview what they uploaded.

As per the user journey the user then needs to confirm the file uploaded is the one they want to use as the initial fact table for the dataset.  If they confirm then we move the file from Blob storage to the Data Lake.